MdeModulePkg/Bus/Usb/UsbBusDxe/UsbHub.c

```diff
@@ -998,6 +998,23 @@ UsbRootHubInit (
 
   if (EFI_ERROR (Status)) {
     gBS->CloseEvent (HubIf->HubNotify);
+    // MU_CHANGE [BEGIN]
+  } else {
+    // MU_CHANGE Add USB Hub Enumeration delay
+    EFI_TPL  OldTpl;
+
+    OldTpl = gBS->RaiseTPL (TPL_HIGH_LEVEL);
+    gBS->RestoreTPL (TPL_APPLICATION);
+
+    while (HubIf->PollCount < USB_ENUM_POLL_MINIMUM_ATTEMPTS) {
+      // wait for a minimum of USB_ENUM_POLL_MINIMUM_ATTEMPTS * (USB_ROOTHUB_POLL_INTERVAL 100 ms)
+    }

```

Comment:
A bit more context could optionally be provided here like:

```suggestion
    while (HubIf->PollCount < USB_ENUM_POLL_MINIMUM_ATTEMPTS) {
      // wait for a minimum of USB_ENUM_POLL_MINIMUM_ATTEMPTS * (USB_ROOTHUB_POLL_INTERVAL 100 ms)
      // with the asynchronous periodic timer interrupt that checks for hub enumeration
      // and updates PollCount on each timer trigger
    }
```